src/hotspot/share/interpreter/oopMapCache.cpp

```diff
@@ -276,26 +276,28 @@ bool OopMapCacheEntry::verify_mask(CellTypeState* vars, CellTypeState* stack, in
 
   // Check if map is generated correctly
   // (Use ?: operator to make sure all 'true' & 'false' are represented exactly the same so we can use == afterwards)
-  Log(interpreter, oopmap) logv;
-  LogStream st(logv.trace());
+  typedef LogTargetImpl<LogLevel::Trace, LOG_TAGS(interpreter, oopmap) > log_type;
+  bool const is_log_enabled = log_type::is_enabled();
+  static const log_type log;
+  LogStream st(log);
```

Comment:
Isn't this just:

```
  const bool log = log_is_enabled(Trace, interpreter, oopmap);
  LogStream st(Log(interpreter, oopmap)::trace());
```

Reply:
Yes, you're right, there are many ways to say the same. Log(...)::trace() returns exactly the object represented by log. May be it should be

`const bool log = log_is_enabled(LogLevel::Trace, interpreter, oopmap);
LogStream st(LogTarget(LogLevel::Trace, interpreter, oopmap));`

Reply:
Or even:

```
LogStreamHandle(Trace, interpreter, oopmap) st;
```

Reply:
LogStream( Log(Trace, interpreter, oopmap)::trace ) seems as the best alternative cuz it does not use confusing "implementation" or "handle" but only public "interfaces"